### DISCUSSION TITLE
Usage of GPIO matrix keypad

### DISCUSSION THREAD

**Discussion by enginlee** [OpenNuvoton/NUC980-linux-4.4.y#78] [Q&A]
Hi,

I am trying to use Linux GPIO matrix keypad driver with my custom NUC980 board.
CONFIG_KEYBOARD_MATRIX has been enabled.
But I don't know how to configure row & column GPIOs.
I followed this document:  https://www.kernel.org/doc/Documentation/devicetree/bindings/input/gpio-matrix-keypad.txt
Here is my device tree:
`		matrix_keypad: matrix-keypad@0 {
			compatible = "gpio-matrix-keypad";
			col-gpios = <
				&gpio 0x46 GPIO_ACTIVE_LOW		/* PC6~PC10 as output */
				&gpio 0x47 GPIO_ACTIVE_LOW
				&gpio 0x48 GPIO_ACTIVE_LOW
				&gpio 0x49 GPIO_ACTIVE_LOW
				&gpio 0x4A GPIO_ACTIVE_LOW
			>;
			row-gpios = <
				&gpio 0xA0 GPIO_ACTIVE_LOW		/* PF0~PF3 as input */
				&gpio 0xA1 GPIO_ACTIVE_LOW
				&gpio 0xA2 GPIO_ACTIVE_LOW
				&gpio 0xA3 GPIO_ACTIVE_LOW
			>;
`
But I got an error when compile DTS file:
syntax error FATAL ERROR: Unable to parse input tree
How do I reference NUC980 GPIOs in device tree?

Thanks,

-Engin

**Comment by yachen** [comment]
dts compiler should report the line that contains syntax error?

**Reply by enginlee** [reply]
Yes, there is an error for parsing "&gpio"
`Error: arch/arm/boot/dts/nuc980-climax-ddr64m-nor32m-kp.dts:442.17-18 syntax error
FATAL ERROR: Unable to parse input tree
scripts/Makefile.lib:293: recipe for target 'arch/arm/boot/dts/nuc980-climax-ddr64m-nor32m-kp.dtb' failed
`
Please check the attachment for the full DTS file
[keypad_dts.txt](https://github.com/OpenNuvoton/NUC980-linux-4.4.y/files/7290237/keypad_dts.txt)

Thank you

**Reply by yachen** [reply]
The problem is not caused by &gpio. It's because of the undefined `GPIO_ACTIVE_LOW`.

**Reply by enginlee** [reply]
Yes, you are right.
After adding `#include <dt-bindings/gpio/gpio.h>` in the front of DTS file, it is working now.
Thank you.
